WordPress/Classes/ViewRelated/Comments/Controllers/Create/CommentCreateViewModel.swift

```diff
@@ -40,7 +40,6 @@ final class CommentCreateViewModel {
     /// Create a reply to the given comment.
     init(replyingTo comment: Comment, save: @escaping (String) async throws -> Void) {
         let siteID = comment.associatedSiteID ?? 0
-        wpAssert(siteID != 0, "missing required parameter siteID")
```

Comment:
Good catch. There are a couple of errors in production (10 to be precise, so very few).

The `siteID` field is used in a couple of places and passing `0` will break `private func makeDraftKey() -> String? {` and potentially break `SuggestionsListViewModel`. This PR needs some additional changes to address the scenarios when `siteID` is missing.

Reply:
`makeDraftKey` returns nil for self-hosted sites. That means it should never get a 0 `siteID`. But I have updated `SuggestionsListViewModel`'s initialiser to accept any `Blog` instance, in https://github.com/wordpress-mobile/WordPress-iOS/pull/24544/commits/f52af5a63e8634e67ca2f2cb9a1b06db10635bc0

Reply:
But there is no reason to support saving drafts only on WP.com sites? I think that should work on all sites?